tests-ng/plugins/dpkg.py

```diff
@@ -26,6 +38,22 @@ def own_and_foreign_architectures(self) -> list[str]:
         architectures.append(self.architecture())
         return architectures
 
+    def collect_packages(self) -> dict[str, str]:
+        """Collect all installed packages and their versions"""
+        result = self._shell(
+            "dpkg-query -W -f='${binary:Package}\t${Version}\n'",
```

Comment:
opinion: in scripts, it is better to use long versions of flags for the sake of readability. I had to look up what `-W` does, and I guess I'm not the only one who does not know by heart

https://manpages.debian.org/trixie/dpkg/dpkg-query.1.en.html#W~2